### PR TITLE
Base the kube2sky container on busybox

### PR DESCRIPTION
**Pull request by thockin**
@satnam6502 @vishh @ArtfulCoder

### CODE REVIEW

**Comment by satnam6502** [comment]
How about a bump on the version number for the TAG in the Makeifle?

**Comment by thockin** [comment]
We have no real release procedure here.  Do we bump the tag whenever we
make a change?  or do we bump it when we make a release?  I assume the
latter, and we're not making a release just for this, right?

**Comment by ArtfulCoder** [comment]
I think for now the procedure should be:
1. make code change.
2. bump tag in Makefile and yaml spec
3. Manually verify
4. checkin 
5. push container with new tag

This will have to continue till we make changes similar to the master components, where we just generate tar files for images as part of build and use those during kube-up and kube-push.


Re-using the same number removes the ability to rollback..

**Comment by satnam6502** [comment]
One reason to bump the tag is that if you make a mistake you can always easily revert to the previous image.

**Comment by vishh** [comment]
I don't think its worth making a release for every PR. It will result in creating large complex PRs. How about building and pushing a kube2sky release as part of the building a kubernetes release?

File: cluster/addons/dns/kube2sky/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM scratch
```
Comment (vishh) [inline cluster/addons/dns/kube2sky/Dockerfile]: How about using [progrium/busybox](https://github.com/progrium/busybox) instead? We use that for cadvisor and heapster and it is very helpful.

Comment (vmarmol) [inline cluster/addons/dns/kube2sky/Dockerfile]: +1 we found that busybox does not have most of the dynamically linked libraries Go uses.

**Comment by satnam6502** [comment]
I'll merge this for now and we can follow up with another PR if the base image should be changed.

File: cluster/addons/dns/kube2sky/Dockerfile (hunk 1 above)
Comment (thockin) [inline cluster/addons/dns/kube2sky/Dockerfile]: progrium is slated for deprecation.  I just want to make the simplest
change I can make right now.

Comment (thockin) [inline cluster/addons/dns/kube2sky/Dockerfile]: Why do you need dynamic libs?  That's sooo 2012.

Comment (vmarmol) [inline cluster/addons/dns/kube2sky/Dockerfile]: You can't actually statically link some dynamic libs dependencies :) we stopped being able to statically link cAdvisor for example.

Comment (thockin) [inline cluster/addons/dns/kube2sky/Dockerfile]: Eh?  Explain?

Comment (vmarmol) [inline cluster/addons/dns/kube2sky/Dockerfile]: Well this case may not be generalized enough :) but we use cgo and with the
latest Go you can't statically link with cgo unless you specify the
compiler and do a few more things. Doing those seemed like a huge pain so
we opted not to. We needed a busybox anyways so we could `docker exec` and
inspect the environment.

Comment (thockin) [inline cluster/addons/dns/kube2sky/Dockerfile]: We link kube binaries statically with `CGO_ENABLED=0 go install
-installsuffix cgo` and it seems to work.  What's a huge pain?

Comment (thockin) [inline cluster/addons/dns/kube2sky/Dockerfile]: To be clear - the fact that some random container happens to have
appropriate libs for you and you DEPEND ON THAT is pretty terrifying.

Comment (satnam6502) [inline cluster/addons/dns/kube2sky/Dockerfile]: There are many terrifying aspects of how we make and deploy and use containers. At some point we should sit down and rationalize it and put in procedures and checks etc.

**Pull request merged**